test/Concurrency/Runtime/isolated_macro_in_nonisolated_nonsending_func.swift

```diff
@@ -0,0 +1,72 @@
+// RUN: %target-run-simple-swift( -Xfrontend -disable-availability-checking -parse-as-library) | %FileCheck %s --dump-input=always
```

Comment:
I kinda would have preferred a test case that checks the SIL of the initializer, but this'll work.